v1/ao/internal/reporter/metrics.go

```diff
@@ -125,13 +122,93 @@ var (
 	cachedContainerID     = "uninitialized" // cached docker container ID (if applicable)
 )
 
-// list of currently stored unique HTTP transaction names (flushed on each metrics report cycle)
-var metricsHTTPTransactions = make(map[string]bool)
+// TransMap records the received transaction names in a metrics report cycle. It will refuse
+// new transaction names if reaching the capacity.
+type TransMap struct {
+	// The map to store transaction names
+	mt map[string]struct{}
```

Comment:
I think abbreviations like `mt` or `mu` are fine for local variables but in a struct  it might be better to give a more descriptive name like `transactionNames` and `mutex`.

Reply:
It makes sense.